polaris-react/src/components/ActionList/components/Item/Item.tsx

```diff
@@ -61,7 +65,15 @@ export function Item({
     );
   }
 
-  const contentText = ellipsis && content ? `${content}…` : content;
+  let contentText = null;
```

Comment:
```js
const contentText = ellipsis && content ? `${content}…` : (truncate ? truncateText(content) : content);
```

Reply:
Thanks for the suggestion. Nesting a ternary operator is shorter, but the `if` `else` style is more readable to other devs.

Reply:
You should consider using const most if not all places, and your condition is duplicated! 

```js
const contextText = content
if (ellipsis && content) {
  contentText = `${content}…`;
} else if (content && truncate) {
  contentText = truncateText(content);
} 
 ```
 
 remove the else, and assign contextText = content as default.
 but i still prefer one line, since its very short condition.

Reply:
We prefer clarity over brevity, especially in this case because theres multiple props that influence  what's rendered and one needs to take precedence. 

```suggestion
  let contentText = content
  if (ellipsis) {
    contentText = `${content}…`;
  } else if (truncate) {
    contentText = truncateText(content);
  } 
```

I do think that `ellipsis` should be[ deprecated](https://github.com/Shopify/polaris/blob/main/documentation/Deprecation%20guidelines.md) in favor of the `truncate` prop you're adding @zakwarsame

Reply:
Working on it [here](https://github.com/Shopify/polaris/pull/9005) 😄